compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3146,14 +3146,21 @@ object Parsers {
         }
         else emptyType
       val rhs =
-        if (tpt.isEmpty || in.token == EQUALS)
+        val deferredOK = !tpt.isEmpty && in.token != EQUALS
```

Comment:
I think the double negation buys us nothing here. I prefer to keep old condition.

Reply:
OK, I'll drop the commit and then squash.